apis/field_error.go

```diff
@@ -50,11 +52,20 @@ func (fe *FieldError) ViaField(prefix ...string) *FieldError {
 	if fe == nil {
 		return nil
 	}
-	var newPaths []string
-	for _, oldPath := range fe.Paths {
-		newPaths = append(newPaths, flatten(append(prefix, oldPath)))
+	var newErrs []FieldError
+	for _, e := range fe.getNormalizedErrors() {
+		// Prepend the Prefix to existing errors.
+		var newPaths []string
+		for _, oldPath := range e.Paths {
+			newPaths = append(newPaths, flatten(append(prefix, oldPath)))
+		}
+		e.Paths = newPaths
+
+		// Append the mutated error to the errors list.
+		newErrs = append(newErrs, e)
 	}
-	fe.Paths = newPaths
+	fe.clear()
```

Comment:
I wonder if these assorted methods should stop mutating themselves and simply return a new `FieldError`.

Reply:
I looked at that and I could make the assertment that a field error is immutable but that was not how the original class was so I made it mutable